### PR TITLE
className property added for input field.

### PR DESCRIPTION
**Pull request by cy6eria**

### CODE REVIEW

**Comment by martijnrusschen** [comment]
Sorry for the delay @cy6eria. I'm really busy this week, so I'll try to get back to you this weekend or early next week! One thing I noticed is that your PR changes 20.000 lines of code, this makes things really hard to review. Is there any way to get your change smaller?

**Comment by iam4x** [comment]
@martijnrusschen This is correct, I've checked it (I need it for my project too)

Look at the end of files changed, he just adds `this.props.className` if it has one.

**Comment by RSO** [comment]
Hey!

Thanks for taking the time to write this! I will look in to this PR first thing tomorrow, as I have the feeling that there might be a better way to do this. I'm also going to look in to a solution to the unreadable PR's because of the big changes in the compiled files.

File: src/datepicker.jsx [hunk 1]
```diff
@@ -77,6 +77,7 @@ var DatePicker = React.createClass({
           setSelected={this.setSelected}
           clearSelected={this.clearSelected}
           hideCalendar={this.hideCalendar}
+          className={this.props.className}
```
Comment (RSO) [inline src/datepicker.jsx]: I think that `className` is not really declarative. Maybe for now we should go with `inputClassName`, unless you have another suggestion?

File: src/date_input.jsx [hunk 2]
```diff
@@ -76,7 +76,7 @@ var DateInput = React.createClass({
       onKeyDown={this.handleKeyDown}
       onFocus={this.props.onFocus}
       onChange={this.handleChange}
-      className="datepicker__input"
+      className={this.props.className ? this.props.className + " datepicker__input" : "datepicker__input"}
```
Comment (RSO) [inline src/date_input.jsx]: How about just defaulting the `className` to `datepicker__input` through `getDefaultProps`? That way we can just do `{this.props.className]` here.

Comment (RSO) [inline src/date_input.jsx]: Otherwise we might want to call this `additionalClassName`.

**Comment by RSO** [comment]
I'm wondering, are error classes the only use case for adding this className? I'm kinda weary of adding classes like this, as is it doesn't really seem to scale that much (e.g. I'm afraid that someone wants a different className for selected days in the future).

We could also think of adding a property that indicates that the datepicker is in an error state.

All in all, if we cant figure out a better way I'd be OK landing this once the comments are addressed, as I don't want to block you for such a small request.

**Comment by martijnrusschen** [comment]
This has been fixed in b61725c82750ac4963c38c12cdc8cbf1c1ded33e, not sure if this was fixed before, but it looks like that this PR isn't needed anymore.

**Pull request closed without merging**